packages/components/src/public/QueryModel/SelectionStatus.tsx

```diff
@@ -53,11 +56,20 @@ export const SelectionStatus: FC<RequiresModelAndActions> = memo(({ actions, mod
         );
     }
 
-    if (rowCount > maxRows && selectionSize !== rowCount && rowCount > 0) {
+    if (
+        rowCount > maxRows &&
+        selectionSize !== rowCount &&
+        rowCount > 0 &&
+        !isLoadingTotalCount &&
+        selectionSize < maxSelectionSize
+    ) {
+        // TODO: Should we clear pagination argument if this is clicked so they go back to the first page if the selection is maxed out?
```

Comment:
I have a sample type with > 500k rows, if I go to the last page and select the 4 rows there and then click the "Select first 100,000", I do think it would make sense to go back to the first page. 

Also, if clicking that select first 100k button surpasses the 100k mark, I get the error state but I can't get out of it without leaving the grid and coming back. Is that expected?

<img width="1617" height="473" alt="Image" src="https://github.com/user-attachments/assets/f38c5691-9db3-4992-a110-be7b2c6e0d62" />